composition/src/linktime_composition.cpp

```diff
@@ -32,7 +32,7 @@ int main(int argc, char * argv[])
   rclcpp::Logger logger = rclcpp::get_logger(LINKTIME_COMPOSITION_LOGGER_NAME);
   rclcpp::executors::SingleThreadedExecutor exec;
   rclcpp::NodeOptions options;
-  std::vector<class_loader::ClassLoader *> loaders;
+  std::vector<std::shared_ptr<class_loader::ClassLoader>> loaders;
```

Comment:
Since there is only ever going to be one owner, I would suggest using a `std::unique_ptr` here.